agent/structs/config_entry.go

```diff
@@ -990,10 +990,10 @@ type PassiveHealthCheck struct {
 	// being removed from the pool.
 	MaxFailures uint32 `json:",omitempty" alias:"max_failures"`
 
-	// EnforcingConsecutive5xx is the % chance that a host will be actually ejected
+	// EnforcingConsecutive5XX is the % chance that a host will be actually ejected
 	// when an outlier status is detected through consecutive 5xx.
 	// This setting can be used to disable ejection or to ramp it up slowly. Defaults to 100.
-	EnforcingConsecutive5xx *uint32 `json:",omitempty" alias:"enforcing_consecutive_5xx"`
+	EnforcingConsecutive5XX *uint32 `json:",omitempty" alias:"enforcing_consecutive_5xx"`
```

Comment:
I renamed this but I'm not sure if it's safe. The reason for name is because when protoc runs, it generates a go struct with the `EnforcingConsecutive5xx` field according to naming conventions which then results in a field named `EnforcingConsecutive5Xx` (notice the capitalized `X` in the `5Xx`). This then results in `mog` failing to convert because the generated go struct doesn’t have field `EnforcingConsecutive5xx`.

@boxofrad let me know what you think.

Reply:
Great call out! To tell you the truth, I'm not entirely sure whether it's safe or not.

I think it probably isn't because it'd change the field name in the messagepack representation we send in net/rpc calls and store in Raft snapshots. You could probably override `MarshalBinary` to get around this, but I'm wondering whether it's worth the risk of regressions.

An alternative workaround would be to manually implement the conversion function for the `PassiveHealthCheck` type, which is a shame for sure, but probably safer.

Reply:
Oh thanks for that information! Yeah, I had an uneasy feeling when I was doing a rename 😄 

I'll do it manually, thanks!

Reply:
Done! Please take a another look when you get a chance!